### PR TITLE
Civ 1270 does your partner receive a pension

### PR DESCRIPTION
**Pull request by laarmada**
### JIRA link ###
https://[tools.hmcts.net](https://tools.hmcts.net/jira/browse/CIV-1270)/jira/browse/CIV-1270

### Change description ###

create a view "does your partner receive a pension?"

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

File: src/main/modules/statementOfMeans/partner/partnerPensionService.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  public async getPartnerPension(claimId: string) {
+    try {
+      const case_data = await getCaseDataFromStore(claimId);
+      if (case_data && case_data.statementOfMeans && case_data.statementOfMeans.partnerPension) {
+        partnerPension.option = case_data.statementOfMeans.partnerPension.option;
+        return partnerPension;
+      }
+      return new PartnerPension();
+    } catch (err: unknown) {
+      logger.error(`${err as Error || err}`);
```
Comment (LeonardoPalmeiro) [inline src/main/modules/statementOfMeans/partner/partnerPensionService.ts:21]: should add   ` res.status(500).send({error: error.message});` to achieve 100% of coverage on test

File: src/main/modules/statementOfMeans/partner/partnerPensionService.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      if (case_data && case_data.statementOfMeans) {
+        case_data.statementOfMeans.partnerPension = partnerPension;
+      } else {
+        const statementOfMeans = new StatementOfMeans();
+        statementOfMeans.partnerPension = partnerPension;
+        case_data.statementOfMeans = statementOfMeans;
+      }
+      await saveDraftClaim(claimId, case_data);
+    } catch (err: unknown) {
+      logger.error(`${err as Error || err}`);
```
Comment (LeonardoPalmeiro) [inline src/main/modules/statementOfMeans/partner/partnerPensionService.ts:37]: should add   ` res.status(500).send({error: error.message});` to achieve 100% of coverage on test

**Review by LeonardoPalmeiro — CHANGES_REQUESTED** [review]

File: src/main/routes/features/response/statementOfMeans/partner/partnerPensionController.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      try {
+        await partnerPensionService.savePartnerPension(req.params.id, partnerPension);
+        const disability = await disabilityService.getDisability(req.params.id);
+        if (disability.option == 'no') {
+          res.redirect(constructResponseUrlWithIdParams(req.params.id, CITIZEN_DEPENDANTS_URL));
+        } else {
+          res.redirect(constructResponseUrlWithIdParams(req.params.id, CITIZEN_PARTNER_DISABILITY_URL));
+        }
+      } catch (err: unknown) {
+        logger.error(`${err as Error || err}`);
```
[the remark this answers is not in the document]
Comment (laarmada) [reply src/main/routes/features/response/statementOfMeans/partner/partnerPensionController.ts]: done

File: src/main/routes/features/response/statementOfMeans/partner/partnerPensionController.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  res.render(citizenPartnerPensionViewPath, {form});
+}
+
+router.get(CITIZEN_PARTNER_PENSION_URL, async (req, res) => {
+  try {
+    const partnerPension = await partnerPensionService.getPartnerPension(req.params.id);
+    renderView(partnerPension, res);
+  } catch (err: unknown) {
+    logger.error(`${err as Error || err}`);
+  }
```
[the remark this answers is not in the document]
Comment (laarmada) [reply src/main/routes/features/response/statementOfMeans/partner/partnerPensionController.ts:28]: done

File: src/main/modules/statementOfMeans/partner/partnerPensionService.ts (hunk 2 above)
Comment (laarmada) [reply src/main/modules/statementOfMeans/partner/partnerPensionService.ts]: done

File: package.json [hunk 5]
```diff
@@ -122,7 +122,7 @@
     "**/jest/**/node-notifier": "^8.0.1",
     "**/sass-lint/**/jsonpointer": "^5.0.0",
     "**/sass-lint/**/merge": "^2.1.1",
-    "**/sass-lint/**/minimist": "^1.2.3",
+    "**/sass-lint/**/minimist": "^1.2.5",
```
Comment (GorilaNaranja) [inline package.json:125]: Why did you update that module? I think Mounika fixed the problem on master upgrading minimist to 1.2.6

**Review by GorilaNaranja — APPROVED** [review]
Nice work!

**Pull request merged**